#install.packages("compbio4all")
#install.packages("rentrez")
#if (!requireNamespace("BiocManager", quietly=TRUE))
 # install.packages("BiocManager")
#BiocManager::install("msa")
#install.packages("texi2dvi")


library(msa)
library(Biostrings)
library(ggplot2)


system.file("tex", "texshade.sty", package="msa")

library(ggmsa)


#prueba

PG_PGRSP <- readAAStringSet("Data/PG_PGRS1/result.fasta")

TB_msa_SP <- msa(PG_PGRSP)

sink("Results/PG_PGRS1/MSA_PE_PGRS1.txt")
print(TB_msa_SP, show="complete")
sink()


class(TB_msa_SP) <- "AAMultipleAlignment"



pdf("Results/PG_PGRS1/Rplotp.pdf", width = 20, height = 10)
# 2. Create the plot
ggmsa(TB_msa_SP, start = 300, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()



#IS10
#GENES PG_PGRS cambios virulentos que tiene


####################################################################################
###################PROTEINAS DE TUBERCULOSIS #######################################
###############################################################################





############ PG_PGRS1 ###################
#

#REPETIR
PG_PGRS1 <- readAAStringSet("Data/PG_PGRS1/sequence.fasta")
names(PG_PGRS1)
PG_PGRS1_A <- PG_PGRS1[-4]
names(PG_PGRS1_A)
TB_msa_S1 <- msa(PG_PGRS1_A)

sink("Results/PG_PGRS1/MSA_PE_PGRS1.txt")
print(TB_msa_S1, show="complete")
sink()


class(TB_msa_S1) <- "AAMultipleAlignment"



pdf("Results/PG_PGRS1/Rplot.pdf", width = 20, height = 10)
# 2. Create the plot
ggmsa(TB_msa_S1, start = 300, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()


 ################## INTERESANTE ##################
############ PG_PGRS2 ########################
# SALEN MUCHA MAS VARIANTES DE BEJING#

PG_PGRS2 <- readAAStringSet("Data/PG_PGRS2/sequence.fasta")



TB_msa_S2 <- msa(PG_PGRS2)

sink("Results/PG_PGRS2_R/MSA_PG_PGRS2.txt")
print(TB_msa_S2, show="complete")
sink()

class(TB_msa_S2) <- "AAMultipleAlignment"

pdf("Results/PG_PGRS2_R/Rplot.pdf", width = 20, height = 10)
# 2. Create the plot
ggmsa(TB_msa_S2, start = 300, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()




#REPETIR
############################## PG_PGRS3 ###################################################3

PG_PGRS3 <- readAAStringSet("Data/PG_PGRS3/sequence (2).fasta")
TB_msa_S3 <- msa(PG_PGRS3)


sink("Results/PG_PGRS3/MSA_PG_PGRS3.txt")
print(TB_msa_S3, show="complete")
sink()

class(TB_msa_S3) <- "AAMultipleAlignment"



pdf("Results/PG_PGRS3/Rplot.pdf", width = 20, height = 10)
# 2. Create the plot
ggmsa(TB_msa_S3, start = 400, end = 500, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()



#REPETIR
##############PG_PGRS4 ######################
#quitar variantes Bovis

PG_PGRS4 <- readAAStringSet("Data/PG_PGRS4/sequence.fasta")
names(PG_PGRS4)
PG_PGRS4_A <- PG_PGRS4[2:4]

TB_msa_S4 <- msa(PG_PGRS4_A)


sink("Results/PG_PGRS4/MSA_PG_PGRS4.txt")
print(TB_msa_S4, show="complete")
sink()


class(TB_msa_S4) <- "AAMultipleAlignment"


pdf("Results/PG_PGRS4/Rplot.pdf", width = 20, height = 10)
# 2. Create the plot
ggmsa(TB_msa_S4, start = 550, end = 600, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()


#REPETIR
############ PG_PGRS5 ###################
PG_PGRS5 <- readAAStringSet("Data/PG_PGRS5/sequence.fasta")
TB_msa_S5 <- msa(PG_PGRS5)

sink("Results/PG_PGRS5/MSA_PG_PGRS5.txt")
print(TB_msa_S5, show="complete")
sink()


class(TB_msa_S5) <- "AAMultipleAlignment"

ggmsa(TB_msa_S5, start = 150, end = 200, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()





#REPETIR
####################  PG_PGRS6 ##################
# HAY DEMASIADAS SECUENCIAS, PERO NO ESPECIFICAN CEPA NI NADA
#SALE CASI IGUAL "PG_PGR FAMILY PROTEIN PG_PGRS6 [M. tuberculosis]



PG_PGRS6 <- readAAStringSet("Data/PG_PGRS6/sequence.fasta")
TB_msa_S6 <- msa(PG_PGRS6)

sink("Results/PG_PGRS6/MSA_PG_PGRS6.txt")
print(TB_msa_S6, show="complete")
sink()


class(TB_msa_S6) <- "AAMultipleAlignment"

ggmsa(TB_msa_S6, start = 100, end = 150, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()



#REPETIR
################# INTERESANTE #####################

############ PG_PGRS7 ###################
#SALEN MUCHAS VARIANTES Y CEPAS MENOS, PERO MAS QUE OTROS

PG_PGRS7 <- readAAStringSet("Data/PG_PGRS7/sequence.fasta")
TB_msa_S7 <- msa(PG_PGRS7)

sink("Results/PG_PGRS7/MSA_PG_PGRS7.txt")
print(TB_msa_S7, show="complete")
sink()


class(TB_msa_S7) <- "AAMultipleAlignment"

ggmsa(TB_msa_S7, start = 950, end = 1000, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()



#REPETIR
############ PG_PGRS8 ###################
#

PG_PGRS8 <- readAAStringSet("Data/PG_PGRS8/sequence.fasta")
TB_msa_S8 <- msa(PG_PGRS8)

sink("Results/PG_PGRS8/MSA_PG_PGRS8.txt")
print(TB_msa_S8, show="complete")
sink()


class(TB_msa_S8) <- "AAMultipleAlignment"

ggmsa(TB_msa_S8, start = 70, end = 130, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()




#REPETIR
############ PG_PGRS9 ###################
#

PG_PGRS9 <- readAAStringSet("Data/PG_PGRS9/sequence.fasta")
TB_msa_S9 <- msa(PG_PGRS9)

sink("Results/PG_PGRS9/MSA_PG_PGRS9.txt")
print(TB_msa_S9, show="complete")
sink()


class(TB_msa_S9) <- "AAMultipleAlignment"

ggmsa(TB_msa_S9, start = 170, end = 230, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()

#REPETIR
############ PG_PGRS10 ###################
#HAY VARIAS DESCRITAS PERO CREO QUE SON LAS MISMAS, CASI SIN CEPAS

PG_PGRS10 <- readAAStringSet("Data/PG_PGRS10/sequence.fasta")
TB_msa_S10 <- msa(PG_PGRS10)

sink("Results/PG_PGRS10/MSA_PG_PGRS10.txt")
print(TB_msa_S10, show="complete")
sink()


class(TB_msa_S10) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS10/R_PLotN123456.pdf", width = 20, height = 5)
# 2. Create the plot
ggmsa(TB_msa_S10, start = 350, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()



#REPETIR
############ PG_PGRS11 ###################
#MUCHAS DE VARIANT BOVIS



PG_PGRS11 <- readAAStringSet("Data/PG_PGRS11/sequence.fasta")
TB_msa_S11 <- msa(PG_PGRS11)

sink("Results/PG_PGRS11/MSA_PE_PGRS11.txt")
print(TB_msa_S11, show="complete")
sink()


class(TB_msa_S11) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS11/Rplot.pdf", width = 20, height = 5)
# 2. Create the plot
ggmsa(TB_msa_S11, start = 175, end = 260, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()


########## A PARTIR DE AQUI SALE MAS VARIANTES ################

#######################################################
############ PG_PGRS12 ###################
#SALEN MUCHAS VARIANTES
#HAY UNA BCG DE MEXICO



PG_PGRS12 <- readAAStringSet("Data/PG_PGRS12/sequence.fasta")
names(PG_PGRS12)
#36, 32, 27,26, 23, 18,17, 1,2
PG_PGRS12_A <- PG_PGRS12[c(-36,-32,-27,-26,-23,-18,-17,-1,-2)]
names(PG_PGRS12_A)  
TB_msa_S12 <- msa(PG_PGRS12_A)

sink("Results/PG_PGRS12/MSA_PE_PGRS12.txt")
print(TB_msa_S12, show="complete")
sink()


class(TB_msa_S12) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS12/Rplot.pdf", width = 30, height = 15)
# 2. Create the plot
ggmsa(TB_msa_S12, start = 1, end = 120, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()


#####################################################
############ PG_PGRS13 ###################
#SALEN MUCHAS VARIANTES, Y OTRAVEZ UNA BCG DE MEXICO



PG_PGRS13 <- readAAStringSet("Data/PG_PGRS13/sequence.fasta")
names(PG_PGRS13)
PG_PGRS13_A <- PG_PGRS13[c(-23,-19,-14, -13, -11, -6, -1,-2,-3)]
names(PG_PGRS13_A)


TB_msa_S13 <- msa(PG_PGRS13_A)

sink("Results/PG_PGRS13/MSA_PE_PGRS13.txt")
print(TB_msa_S13, show="complete")
sink()


class(TB_msa_S13) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS13/Rplot.pdf", width = 30, height = 15)
# 2. Create the plot
ggmsa(TB_msa_S13, start = 700, end = 800, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()







###############################################
############ PG_PGRS14 ###################
#SALEN MUCHAS VARIANTES, Y OTRAVEZ UNA BCG DE MEXICO



PG_PGRS14 <- readAAStringSet("Data/PG_PGRS14/sequence.fasta")
names(PG_PGRS14)
#46, 44, 42, 38, 35,1,2
PG_PGRS14_A <- PG_PGRS14[c(-46,-44,-42,-38,-35,-1,-2)]
names(PG_PGRS14_A)

TB_msa_S14 <- msa(PG_PGRS14_A)




sink("Results/PG_PGRS14/MSA_PE_PGRS14.txt")
print(TB_msa_S14, show="complete")
sink()


class(TB_msa_S14) <- "AAMultipleAlignment"



pdf("Results/PG_PGRS14/Rplot.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(TB_msa_S14, start = 220, end = 300, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()







############ PG_PGRS15 ###################
#SALEN MUCHAS VARIANTES

PG_PGRS15 <- readAAStringSet("Data/PG_PGRS15/sequence.fasta")
names(PG_PGRS15)
#25, 21,12,13, 9, 5, 4, 1
PG_PGRS15_A <- PG_PGRS15[c(-25,-21,-12,-13,-9,-5,-4,-1)]
names(PG_PGRS15_A)


TB_msa_S15 <- msa(PG_PGRS15_A)

sink("Results/PG_PGRS15/MSA_PE_PGRS15.txt")
print(TB_msa_S15, show="complete")
sink()


class(TB_msa_S15) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS15/Rplot.pdf", width = 30, height = 15)
# 2. Create the plot
ggmsa(TB_msa_S15, start = 50, end = 150, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()






############ PG_PGRS16 ###################
#


PG_PGRS16 <- readAAStringSet("Data/PG_PGRS16/sequence.fasta")
names(PG_PGRS16)
#39,38,28,22, 1
PG_PGRS16_A <- PG_PGRS16[c(-39, -38,-28,-22,-1)]
names(PG_PGRS16_A)

TB_msa_S16 <- msa(PG_PGRS16_A)

sink("Results/PG_PGRS16/MSA_PE_PGRS16.txt")
print(TB_msa_S16, show="complete")
sink()


class(TB_msa_S16) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS16/Rplot.pdf", width = 30, height = 15)
# 2. Create the plot
ggmsa(TB_msa_S16, start = 300, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()




###########################SOLO HAY DATOS, NO ANALISIS######################

#22 de enero del 2024
############ PG_PGRS17 ###################
#


PG_PGRS16 <- readAAStringSet("Data/PG_PGRS16/sequence.fasta")
names(PG_PGRS16)


TB_msa_S16 <- msa(PG_PGRS16_A)

sink("Results/PG_PGRS16/MSA_PE_PGRS16.txt")
print(TB_msa_S16, show="complete")
sink()


class(TB_msa_S16) <- "AAMultipleAlignment"

#AVR SI ASI SE VEN LOS NOMBRES
#ASI SE VE MUY BONITO width = 30, height =10
#ASI SE VE MEJOR width = 20, height = 5)

pdf("Results/PG_PGRS16/Rplot.pdf", width = 30, height = 15)
# 2. Create the plot
ggmsa(TB_msa_S16, start = 300, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()
























######################## PE_PGRS33 #####################


PG_PGRS33 <- readAAStringSet("Data/PG_PGRS33/sequence.fasta")



TB_msa_S33 <- msa(PG_PGRS33)

sink("Results/PG_PGRS33/MSA_PE_GRS33a.txt")
print(TB_msa_S33, show="complete")
sink()


class(TB_msa_S33) <- "AAMultipleAlignment"


pdf("Results/PG_PGRS33/rplot.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(TB_msa_S33, start = 370, end = 470, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()

###################


########### # PPE2 #####################################
#ESTA SUPER CONSERVADAAAAAAAAAA

PPE2 <- readAAStringSet("Data/PPE2/sequencePPE2.fasta")



PPE2_MSA <- msa(PPE2)

sink("Results/PPE2_MSA/PPE2MSA.txt")
print(PPE2_MSA, show="complete")
sink()


class(PPE2_MSA) <- "AAMultipleAlignment"


pdf("Results/PPE2_MSA/rplot1.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(PPE2_MSA, start = 1, end = 550, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()



##################################
###########PE_PGRS62##############
###### TAMBIEN MUY CONSERVADA ###############


PE_PGRS62 <- readAAStringSet("Data/PE_PGRS62/sequence (2).fasta")

PE_PGRS62_MSA <- msa(PE_PGRS62)

sink("Results/PE_PGRS62/MSA_PE_PGRS62.txt")
print(PE_PGRS62_MSA, show="complete")
sink()


class(PE_PGRS62_MSA) <- "AAMultipleAlignment"


pdf("Results/PE_PGRS62/rplot.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(PE_PGRS62_MSA, start = 200, end = 300, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()



###################################################
########### PE5 #############################
###################MUY CONSERVADAAAAAAAA ####

PE_5 <- readAAStringSet("Data/PE5/sequence (2).fasta")

PE_5_MSA <- msa(PE_5)

sink("Results/PE5/MSA_PEE5.txt")
print(PE_5_MSA, show="complete")
sink()


class(PE_5_MSA) <- "AAMultipleAlignment"


pdf("Results/PE5/rplot.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(PE_5_MSA, start = 1, end = 100, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()


###################################
#############  PPE15 ##################


PPE15 <- readAAStringSet("Data/PPE15/sequence (2).fasta")

PPE15_MSA <- msa(PPE15)

sink("Results/PPE15/PPE15_MSA.txt")
print(PPE15_MSA, show="complete")
sink()


class(PPE15_MSA) <- "AAMultipleAlignment"


pdf("Results/PPE15/rplotT.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(PPE15_MSA, start = 1, end = 400, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()



##################### PE_PGRS47 ##############

PE_PGRS47 <- readAAStringSet("Data/PE_PGRS47/sequence (2).fasta")

PE_PGRS47_MSA <- msa(PE_PGRS47)

sink("Results/PE_PGRS47/MSA_PE_PGRS47.txt")
print(PE_PGRS47_MSA, show="complete")
sink()


class(PE_PGRS47_MSA) <- "AAMultipleAlignment"


pdf("Results/PE_PGRS47/rplotT.pdf", width = 40, height = 20)
# 2. Create the plot
ggmsa(PE_PGRS47_MSA, start = 1, end = 550, char_width = 0.5, seq_name = T)+
  geom_seqlogo()+
  geom_msaBar()
# 3. Close the file
dev.off()









